In [1]:
import pandas as pd
import json

# --- 1. Load your JSON ---
# Replace 'your_file.json' with the path to your JSON file
with open('data/LesOutputJson.json', encoding='utf-8') as f:
    json_data = json.load(f)

data = json_data['Data']

In [2]:
# --- 2. Extract main company info ---
main_cols = [
    "Регистрийн дугаар",
    "Оноосон нэр",
    "Бүртгэсэн огноо",
    "Хэлбэр",
    "Төрөл"
]

df_company = pd.DataFrame([row[:5] for row in data], columns=main_cols)

# --- Helper function to extract nested arrays ---
def extract_nested(data, index, columns):
    rows = []
    for row in data:
        reg = row[0]  # Регистрийн дугаар
        nested_list = row[index] if row[index] else []
        for item in nested_list:
            record = {"Регистрийн дугаар": reg}
            record.update({col: item[i] for i, col in enumerate(columns)})
            rows.append(record)
    return pd.DataFrame(rows)

# --- 3. Shareholders ---
shareholder_cols = ["Ангилал", "Улсын нэр", "Эцэг/эх/-ийн нэр", "Нэр", "Хүйс", "Бүртгэсэн огноо"]
df_shareholders = extract_nested(data, 5, shareholder_cols)

# --- 4. Representatives ---
rep_cols = ["Албан тушаал", "Улсын нэр", "Эцэг/эх/-ийн нэр", "Нэр", "Хүйс", "Бүртгэсэн огноо"]
df_representatives = extract_nested(data, 6, rep_cols)

# --- 5. Final owners ---
owner_cols = ["Ангилал", "Эцэг/эх/-ийн нэр", "Нэр", "Хүйс", "Бүртгэсэн огноо"]
df_final_owners = extract_nested(data, 7, owner_cols)

# --- 6. Business activities ---
activity_cols = ["Төлөв", "Бүртгэсэн огноо", "Үйл ажиллагааны чиглэл"]
df_activities = extract_nested(data, 8, activity_cols)

# --- 7. Reorganization history ---
reorg_cols = ["Өөрчлөгдөн зохион байгуулсан хэлбэр", "Бүртгэсэн огноо", "Өөрчлөлтийн өмнөх оноосон нэр", "Өөрчлөлтийн тэмдэглэл"]
df_reorg = extract_nested(data, 9, reorg_cols)

# --- 8. Quick check ---
print("=== Company Table ===")
print(df_company.head())
print("\n=== Shareholders ===")
print(df_shareholders.head())
print("\n=== Representatives ===")
print(df_representatives.head())
print("\n=== Final Owners ===")
print(df_final_owners.head())
print("\n=== Activities ===")
print(df_activities.head())
print("\n=== Reorganizations ===")
print(df_reorg.head())

# ✅ All DataFrames ready for analysis


=== Company Table ===
  Регистрийн дугаар           Оноосон нэр Бүртгэсэн огноо  \
0           3199665       Ташир агро фийд      2021.03.05   
1           6673252                  РБМК      2021.03.05   
2           6678912     Уудам баян бөөрөг      2021.03.16   
3           6678971  Хан технологийн гүүр      2021.03.16   
4           6678904     Нүүдэлчин их оньс      2021.03.16   

                                 Хэлбэр             Төрөл  
0  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
1  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
2  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
3  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
4                        Холбоо /НҮТББ/  Ашгийн төлөө бус  

=== Shareholders ===
  Регистрийн дугаар Ангилал   Улсын нэр Эцэг/эх/-ийн нэр         Нэр Хүйс  \
0           3199665   Иргэн  Монгол Улс       даваасүрэн  эрдэнэтуяа   Эм   
1           6673252   Иргэн  Монгол Улс      цэдэнбалжир  идэрбүргэд   Эр  

In [3]:
df_company['Регистрийн дугаар'].value_counts()

Регистрийн дугаар
7083889    2
6807941    2
4419154    2
4112873    2
6808263    2
          ..
1044109    1
8018154    1
1042157    1
3076016    1
3449181    1
Name: count, Length: 275295, dtype: int64

In [4]:
df_company.shape

(275342, 5)

In [5]:
df_company = df_company.drop_duplicates()
df_company.shape

(275295, 5)

# Data cleaning

Company main info table -> Raw: 275,342 rows -> Drop duplicates -> 275,295 rows
Shareholders info table -> Raw: 633,515 rows -> Drop duplicates -> 623,398 rows
Representitives info table -> Raw: 245,917 rows -> Drop duplicates -> 245,872 rows
Final owners info table -> Raw: 326,128 -> Drop duplicates -> 326,041 rows
Activities info table -> Raw: 1,013,536 -> Drop duplicates -> 1,011,030 rows
Company reorganization info table -> Raw: 543 rows -> Drop duplicates -> 539 rows

In [6]:
df_shareholders.shape

(633515, 7)

In [7]:
df_shareholders

Регистрийн дугаар   Ангилал   Улсын нэр Эцэг/эх/-ийн нэр          Нэр  \
0                3199665     Иргэн  Монгол Улс       даваасүрэн   эрдэнэтуяа   
1                6673252     Иргэн  Монгол Улс      цэдэнбалжир   идэрбүргэд   
2                6678912     Иргэн  Монгол Улс       оюундэлгэр    амарсанаа   
3                6678971     Иргэн  Монгол Улс      дамбадаржаа       алимаа   
4                6678939     Иргэн  Монгол Улс         уртнасан   болорчимэг   
...                  ...       ...         ...              ...          ...   
633510           3783537     Иргэн  Монгол Улс        эрхэмбаяр  мөнгөншагай   
633511           7182785     Иргэн  Монгол Улс         батбилэг        түшиг   
633512           3404757  Х/Этгээд  Монгол Улс                      Бэхгүрэн   
633513           7182985     Иргэн  Монгол Улс          ганболд   мягмардорж   
633514           3449181     Иргэн  Монгол Улс        сандуйжав     өлзийбат   

       Хүйс Бүртгэсэн огноо  
0        Эм      2021.03.05  
1        Эр      2021.03.05  
2        Эр      2021.03.16  
3        Эм      2023.04.14  
4        Эм      2024.05.22  
...     ...             ...  
633510   Эр      2025.08.13  
633511   Эр      2025.08.13  
633512           2025.08.13  
633513   Эр      2025.08.13  
633514   Эр      2025.08.14  

[633515 rows x 7 columns]

In [8]:
df_representatives

Регистрийн дугаар        Албан тушаал   Улсын нэр Эцэг/эх/-ийн нэр  \
0                3199665  Гүйцэтгэх  захирал  Монгол Улс       Даваасүрэн   
1                6673252  Гүйцэтгэх  захирал  Монгол Улс      Цэдэнбалжир   
2                6678912  Гүйцэтгэх  захирал  Монгол Улс       Оюундэлгэр   
3                6678971  Гүйцэтгэх  захирал  Монгол Улс      Дамбадаржаа   
4                6678904   Гүйцэтгэх захирал  Монгол Улс           Нэргүй   
...                  ...                 ...         ...              ...   
245912           7182785  Гүйцэтгэх  захирал  Монгол Улс         Батбилэг   
245913           3404757  Гүйцэтгэх  захирал  Монгол Улс        Батзоригт   
245914           7182985  Гүйцэтгэх  захирал  Монгол Улс          Ганболд   
245915           3267989   Гүйцэтгэх захирал  Монгол Улс            Ойдов   
245916           3449181  Гүйцэтгэх  захирал  Монгол Улс        Сандуйжав   

               Нэр Хүйс Бүртгэсэн огноо  
0       Эрдэнэтуяа   Эм      2021.03.05  
1       Идэрбүргэд   Эр      2021.03.05  
2        Амарсанаа   Эр      2021.03.16  
3           Алимаа   Эм      2023.04.14  
4       Балдандорж   Эр      2021.03.16  
...            ...  ...             ...  
245912       Түшиг   Эр      2025.08.13  
245913     Батбаяр   Эр      2025.08.13  
245914  Мягмардорж   Эр      2025.08.13  
245915   Батжаргал   Эм      2025.08.14  
245916    Өлзийбат   Эр      2025.08.14  

[245917 rows x 7 columns]

In [9]:
df_representatives.shape

(245917, 7)

In [10]:
df_representatives = df_representatives.drop_duplicates()
df_representatives.shape

(245872, 7)

In [11]:
df_final_owners

Регистрийн дугаар Ангилал Эцэг/эх/-ийн нэр          Нэр        Хүйс  \
0                6673252   Иргэн       Идэрбүргэд  Цэдэнбалжир  Монгол Улс   
1                6678912   Иргэн        Амарсанаа   Оюундэлгэр  Монгол Улс   
2                6678971   Иргэн           Алимаа  Дамбадаржаа  Монгол Улс   
3                6678904   Иргэн       Балдандорж       Нэргүй  Монгол Улс   
4                6678939   Иргэн       Болорчимэг     Уртнасан  Монгол Улс   
...                  ...     ...              ...          ...         ...   
326123           7182785   Иргэн            Түшиг     Батбилэг  Монгол Улс   
326124           3404757   Иргэн          Батбаяр    Батзоригт  Монгол Улс   
326125           7182985   Иргэн       Мягмардорж      Ганболд  Монгол Улс   
326126           3267989   Иргэн       Сарангэрэл  Няндагцэрэн  Монгол Улс   
326127           3449181   Иргэн         Өлзийбат    Сандуйжав  Монгол Улс   

       Бүртгэсэн огноо  
0                   Эр  
1                   Эр  
2                   Эм  
3                   Эр  
4                   Эм  
...                ...  
326123              Эр  
326124              Эр  
326125              Эр  
326126              Эм  
326127              Эр  

[326128 rows x 6 columns]

In [12]:
df_final_owners.shape

(326128, 6)

In [13]:
df_final_owners = df_final_owners.drop_duplicates()
df_final_owners.shape

(326041, 6)

In [14]:
df_activities

Регистрийн дугаар Төлөв Бүртгэсэн огноо  \
0                 3199665  Тийм      2021.03.05   
1                 3199665  Тийм      2021.03.05   
2                 6673252  Тийм      2021.03.05   
3                 6673252  Тийм      2021.03.05   
4                 6673252  Тийм      2021.03.05   
...                   ...   ...             ...   
1013531           7182985  Тийм      2025.08.13   
1013532           3267989  Тийм      2025.08.14   
1013533           3449181  Тийм      2025.08.14   
1013534           3449181  Тийм      2025.08.14   
1013535           3449181  Тийм      2025.08.14   

                                    Үйл ажиллагааны чиглэл  
0        Газар тариалан /Үр тариа, төмс хүнсний ногоо, ...  
1                                              Мал аж ахуй  
2                            Төмөр хашаа, хайс, хаалт хийх  
3                                     Хэвлэлийн эх бэлтгэх  
4                             Барилгын материалын худалдаа  
...                                                    ...  
1013531                                     Аялал жуулчлал  
1013532  Хүүхэд гэр бүл, албан байгууллага сэтгэлзүйн з...  
1013533  Төлбөр эсвэл гэрээний үндсэн дээр хийгдэх бөөн...  
1013534                         Хөгжмийн зэмсэг үйлдвэрлэл  
1013535                  Морин хуурын үйлдвэрлэл, худалдаа  

[1013536 rows x 4 columns]

In [15]:
df_activities.shape

(1013536, 4)

In [16]:
df_activities = df_activities.drop_duplicates()
df_activities.shape

(1011030, 4)

In [17]:
df_reorg

Регистрийн дугаар Өөрчлөгдөн зохион байгуулсан хэлбэр Бүртгэсэн огноо  \
0             5841127                             Нэгтгэх       11-APR-16   
1             5841127                             Нэгтгэх       12-MAY-17   
2             5841127                             Нэгтгэх       20-JAN-20   
3             5414105                             Нэгтгэх       24-DEC-20   
4             5414105                             Нэгтгэх       24-MAR-21   
..                ...                                 ...             ...   
538           6577539                             Нэгтгэх       30-DEC-21   
539           6577539                             Нэгтгэх       30-DEC-21   
540           3897907                        Шинэ бүртгэл       03-SEP-25   
541           4040694                           Нийлүүлэх       27-NOV-25   
542           7167737                        Шинэ бүртгэл       12-JUN-25   

                         Өөрчлөлтийн өмнөх оноосон нэр  \
0                                      САТАС корпораци   
1                                      САТАС корпораци   
2                                      САТАС корпораци   
3                                          Нэмэхдэвшил   
4                                          Нэмэхдэвшил   
..                                                 ...   
538                      Иргэний нисэхийн үндэсний төв   
539                      Иргэний нисэхийн үндэсний төв   
540  Төв аймгийн Зуунмод сумын ерөнхий боловсролын ...   
541              Баруун бүсийн цахилгаан түгээх сүлжээ   
542                                  Таван богд сервис   

                                 Өөрчлөлтийн тэмдэглэл  
0    5384621 тоот регистрийн дугаартай Сатапартнерс...  
1    5440866 тоот регистрийн дугаартай Сатакашемир ...  
2    5669928 тоот регистрийн дугаартай Оргилгурван ...  
3    3859436 тоот регистрийн дугаартай Жэй Эй Икс-г...  
4    3859436 тоот регистрийн дугаартай Жэй Эй Икс-г...  
..                                                 ...  
538  5055954 тоот регистрийн дугаартай Буянт-Ухаа о...  
539  5055954 тоот регистрийн дугаартай Буянт-Ухаа о...  
540                                                     
541  2684853 тоот регистрийн дугаартай Ховдцахилгаа...  
542                                                     

[543 rows x 5 columns]

In [18]:
df_reorg = df_reorg.drop_duplicates()
df_reorg.shape

(539, 5)

In [61]:
df_shareholders[(df_shareholders['Эцэг/эх/-ийн нэр']=='ганбат')&(df_shareholders['Нэр']=='чинзоригт')]

Empty DataFrame
Columns: [Регистрийн дугаар, Ангилал, Улсын нэр, Эцэг/эх/-ийн нэр, Нэр, Хүйс, Бүртгэсэн огноо]
Index: []

In [64]:
df_shareholders[df_shareholders['Регистрийн дугаар']==2693321]

Empty DataFrame
Columns: [Регистрийн дугаар, Ангилал, Улсын нэр, Эцэг/эх/-ийн нэр, Нэр, Хүйс, Бүртгэсэн огноо]
Index: []

In [68]:
df_company[df_company['Регистрийн дугаар']==2693321]

Empty DataFrame
Columns: [Регистрийн дугаар, Оноосон нэр, Бүртгэсэн огноо, Хэлбэр, Төрөл]
Index: []

In [67]:
df_company

Регистрийн дугаар            Оноосон нэр Бүртгэсэн огноо  \
0                3199665        Ташир агро фийд      2021.03.05   
1                6673252                   РБМК      2021.03.05   
2                6678912      Уудам баян бөөрөг      2021.03.16   
3                6678971   Хан технологийн гүүр      2021.03.16   
4                6678904      Нүүдэлчин их оньс      2021.03.16   
...                  ...                    ...             ...   
275337           7182785  Эгдүүтэй саарал бүжин      2025.08.13   
275338           3404757             Степ гласс      2025.08.13   
275339           7182985          Бадрах соёмбо      2025.08.13   
275340           3267989                Аюурлан      2025.08.14   
275341           3449181      Яруухан дуут хуур      2025.08.14   

                                      Хэлбэр             Төрөл  
0       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
1       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
2       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
3       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
4                             Холбоо /НҮТББ/  Ашгийн төлөө бус  
...                                      ...               ...  
275337  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
275338  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
275339  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
275340                        Холбоо /НҮТББ/  Ашгийн төлөө бус  
275341  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  

[275295 rows x 5 columns]

In [72]:
df_company['Бүртгэсэн огноо'].min()

''

In [76]:
df_company = df_company[df_company['Бүртгэсэн огноо']!='']

In [77]:
df_company

Регистрийн дугаар            Оноосон нэр Бүртгэсэн огноо  \
0                3199665        Ташир агро фийд      2021.03.05   
1                6673252                   РБМК      2021.03.05   
2                6678912      Уудам баян бөөрөг      2021.03.16   
3                6678971   Хан технологийн гүүр      2021.03.16   
4                6678904      Нүүдэлчин их оньс      2021.03.16   
...                  ...                    ...             ...   
275337           7182785  Эгдүүтэй саарал бүжин      2025.08.13   
275338           3404757             Степ гласс      2025.08.13   
275339           7182985          Бадрах соёмбо      2025.08.13   
275340           3267989                Аюурлан      2025.08.14   
275341           3449181      Яруухан дуут хуур      2025.08.14   

                                      Хэлбэр             Төрөл  
0       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
1       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
2       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
3       Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
4                             Холбоо /НҮТББ/  Ашгийн төлөө бус  
...                                      ...               ...  
275337  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
275338  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
275339  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  
275340                        Холбоо /НҮТББ/  Ашгийн төлөө бус  
275341  Хязгаарлагдмал хариуцлагатай компани      Ашгийн төлөө  

[275294 rows x 5 columns]

In [78]:
df_company['Бүртгэсэн огноо'].min()

'1926.08.20'

In [79]:
df_company['Бүртгэсэн огноо'].max()

'2025.12.15'

In [81]:
df_company.sort_values('Бүртгэсэн огноо')[:20]

Регистрийн дугаар                                        Оноосон нэр  \
136385           9029362                     Дорнод аймгийн тагнуулын газар   
179971           1800094              Олон улсын агаарын тээврийн Эйр Чайна   
152235           1800108                                 Сумитомо корпораци   
149092           1800582                                 Марубени корпораци   
92411            2159759                                            Уурхайт   
148566           1801333                                      Эсүандоэсү ХК   
154384           1801589                                  Европ Трейдинг БВ   
240380           1801562                             Рэдбурне Гоулд компани   
155329           1801651  Аэрофлот-Оросын олон улсын агаарын зам хувьцаа...   
150152           1802534                                 Мицубиши корпораци   
1430             9920137  Өнгөт төмөрлөгийн гадаад барилга зураг төслийн...   
153261           1802623                            Кока-Кола Чайна лимитед   
75700            2684594                                               ТУБО   
92409            2241145                                        Манхантулга   
130366           2782766                                         Хос дэвэлт   
88277            3737357                                         Бүтээлчхүү   
166702           9139176                                Фьючер энтерпрайзис   
149088           9139265  Хилийн чанадад байгаа төрийн өмч хөрөнгийг уди...   
101044           2798204                                                ПАЖ   
86101            2661187                                           Нур-Едил   

       Бүртгэсэн огноо                                         Хэлбэр  \
136385      1926.08.20                Төрийн байгууллага, албан газар   
179971      1991.12.19  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
152235      1991.12.25  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
149092      1994.02.24  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
92411       1996.02.07                                 Нөхөрлөл /ЗБН/   
148566      1996.10.04  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
154384      1997.04.18  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
240380      1997.04.18  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
155329      1997.05.17  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
150152      2000.02.16  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
1430        2000.07.03  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
153261      2001.11.01  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
75700       2002.10.07           Хязгаарлагдмал хариуцлагатай компани   
92409       2003.02.20           Хязгаарлагдмал хариуцлагатай компани   
130366      2003.10.20           Хязгаарлагдмал хариуцлагатай компани   
88277       2004.01.23           Хязгаарлагдмал хариуцлагатай компани   
166702      2004.03.11  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
149088      2004.04.29  Гадаадын хуулийн этгээдийн төлөөлөгчийн газар   
101044      2004.07.06           Хязгаарлагдмал хариуцлагатай компани   
86101       2004.08.26           Хязгаарлагдмал хариуцлагатай компани   

                   Төрөл  
136385  Ашгийн төлөө бус  
179971  Ашгийн төлөө бус  
152235  Ашгийн төлөө бус  
149092  Ашгийн төлөө бус  
92411       Ашгийн төлөө  
148566  Ашгийн төлөө бус  
154384  Ашгийн төлөө бус  
240380  Ашгийн төлөө бус  
155329  Ашгийн төлөө бус  
150152  Ашгийн төлөө бус  
1430    Ашгийн төлөө бус  
153261  Ашгийн төлөө бус  
75700       Ашгийн төлөө  
92409       Ашгийн төлөө  
130366      Ашгийн төлөө  
88277       Ашгийн төлөө  
166702  Ашгийн төлөө бус  
149088  Ашгийн төлөө бус  
101044      Ашгийн төлөө  
86101       Ашгийн төлөө

In [88]:
df_company[df_company['Бүртгэсэн огноо']=='2005.11.04'].sort_values('Оноосон нэр', ascending=False)[:30]

Регистрийн дугаар                                  Оноосон нэр  \
71375            2725533                                      Өсөхлуу   
72041            4368002                                   Өрлөгзэлмэ   
97869            2564394                                     Эххөгжил   
93464            5027462                                    Энх-Асрал   
94918            5027616                                     Эксэфийл   
76167            2622416                                 Цэнгүүндалай   
7957             5027675                                     Холистик   
158218           5027667                                     Хасхүлэг   
128584           2693321                                    ХААН банк   
10295            5027713                               Фрэндшип наран   
11298            2556294                                 Стар-Эстэйтс   
71736            4367979                                        Сорьж   
10299            5027543                                        Пи Эм   
72045            2691043                                   Оюут-Өргөө   
116365           4367987  Орхон аймгийн хөдөө аж ахуйн нэвтрүүлэх төв   
74870            2012537                                  Онгийн хийд   
74011            2639904                             Онги есөн эрдэнэ   
71731            4367952                                    Ноёнсанаа   
172251           5027659                                   Нарны орон   
115880           2239205                                          НБЧ   
115879           2123711                                Мөнхтогтох-Од   
71599            2775565                             Мөнх ундрам хүнс   
158373           5027551                                        Мүгэн   
8492             2564351                        Монголиа траст венчур   
10300            5027519                                        Мокон   
10297            5027608                                       Милтон   
93468            5027683                                   Кэй Жи Эйч   
10296            5027624                                       Кардио   
71740            4367995                                          КСМ   
70673            2823802                                      Зөөхэрд   

       Бүртгэсэн огноо                                             Хэлбэр  \
71375       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
72041       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
97869       2005.11.04  Гадаадын хөрөнгө оруулалттай хязгаарлагдмал ха...   
93464       2005.11.04  Гадаадын хөрөнгө оруулалттай хязгаарлагдмал ха...   
94918       2005.11.04  Гадаадын хөрөнгө оруулалттай хязгаарлагдмал ха...   
76167       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
7957        2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
158218      2005.11.04  Гадаадын хөрөнгө оруулалттай хязгаарлагдмал ха...   
128584      2005.11.04                     Хувьцаат компани /нээлттэй ХК/   
10295       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
11298       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
71736       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
10299       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
72045       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
116365      2005.11.04  Төрийн байгууллага, албан газар, улсын төсөвт ...   
74870       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
74011       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
71731       2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
172251      2005.11.04               Хязгаарлагдмал хариуцлагатай компани   
115880      2005.11.04                                     Нөхөрлөл /ЗБН/   
115879      2005.11.04                                     Нөхөрлөл /ЗБН/   
71599       2005.11.04               Хязгаарлагдм

In [90]:
df_company[df_company['Регистрийн дугаар']=='2693321']

Регистрийн дугаар Оноосон нэр Бүртгэсэн огноо  \
128584           2693321   ХААН банк      2005.11.04   

                                Хэлбэр         Төрөл  
128584  Хувьцаат компани /нээлттэй ХК/  Ашгийн төлөө

In [92]:
df_shareholders[df_shareholders['Регистрийн дугаар']=='2693321']

Регистрийн дугаар Ангилал   Улсын нэр Эцэг/эх/-ийн нэр          Нэр  \
288958           2693321   Иргэн  Монгол Улс         дашдаваа        хулан   
288959           2693321   Иргэн  Монгол Улс             дияр      еркебек   
288960           2693321   Иргэн  Монгол Улс        эрдэнэбат   ууганчимэг   
288961           2693321   Иргэн  Монгол Улс        адъяадорж   эрдэнэмөнх   
288962           2693321   Иргэн  Монгол Улс      даринчулуун  долгорсүрэн   
...                  ...     ...         ...              ...          ...   
293095           2693321   Иргэн  Монгол Улс        юндэндорж       энхзул   
293096           2693321   Иргэн  Монгол Улс          энхболд    батэрдэнэ   
293097           2693321   Иргэн  Монгол Улс        баясгалан      билгүүн   
293098           2693321   Иргэн  Монгол Улс        алтантуяа     ганчимэг   
293099           2693321   Иргэн  Монгол Улс          ваанчиг      солонго   

       Хүйс Бүртгэсэн огноо  
288958   Эм      2022.09.06  
288959   Эр      2022.09.06  
288960   Эм      2022.09.06  
288961   Эр      2022.09.06  
288962   Эм      2022.09.06  
...     ...             ...  
293095   Эм      2022.09.06  
293096   Эр      2022.09.06  
293097   Эр      2022.09.06  
293098   Эм      2022.09.06  
293099   Эм      2022.09.06  

[4107 rows x 7 columns]

In [97]:
df_filter = df_shareholders[(df_shareholders['Эцэг/эх/-ийн нэр']=='дашдаваа')&(df_shareholders['Нэр']=='хулан')]
df_filter

Регистрийн дугаар Ангилал   Улсын нэр Эцэг/эх/-ийн нэр    Нэр Хүйс  \
2168             6643744   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
12411            2543303   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
170873           5803403   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
180446           2160544   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
288958           2693321   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
415022           5646804   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
521121           7002913   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   

       Бүртгэсэн огноо  
2168        2020.12.16  
12411       2022.11.28  
170873      2022.11.25  
180446      2024.02.02  
288958      2022.09.06  
415022      2018.01.16  
521121      2024.03.15

In [100]:
pd.merge(df_filter, df_company, on="Регистрийн дугаар", how="inner")

Регистрийн дугаар Ангилал   Улсын нэр Эцэг/эх/-ийн нэр    Нэр Хүйс  \
0           6643744   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
1           2543303   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
2           5803403   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
3           2160544   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
4           2693321   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
5           5646804   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   
6           7002913   Иргэн  Монгол Улс         дашдаваа  хулан   Эм   

  Бүртгэсэн огноо_x             Оноосон нэр Бүртгэсэн огноо_y  \
0        2020.12.16          Ар И Ди эженси        2020.12.16   
1        2022.11.28     Таван богд холдингс        2006.01.13   
2        2022.11.25  Таван богд фүүдс пицца        2014.02.20   
3        2024.02.02               Буянт-Өрх        2006.05.24   
4        2022.09.06               ХААН банк        2005.11.04   
5        2018.01.16         Таванбогд фүүдс        2012.11.01   
6        2024.03.15  Вайт маркетинг эйженси        2024.03.15   

                                 Хэлбэр         Төрөл  
0  Хязгаарлагдмал хариуцлагатай компани  Ашгийн төлөө  
1  Хязгаарлагдмал хариуцлагатай компани  Ашгийн төлөө  
2  Хязгаарлагдмал хариуцлагатай компани  Ашгийн төлөө  
3                        Нөхөрлөл /ЗБН/  Ашгийн төлөө  
4        Хувьцаат компани /нээлттэй ХК/  Ашгийн төлөө  
5  Хязгаарлагдмал хариуцлагатай компани  Ашгийн төлөө  
6  Хязгаарлагдмал хариуцлагатай компани  Ашгийн төлөө

In [19]:
df_company.to_csv("data/company_main_info.csv", encoding='utf-8-sig',index=False)

In [20]:
df_shareholders.to_csv("data/shareholders_info.csv", encoding='utf-8-sig',index=False)

In [21]:
df_representatives.to_csv("data/reprentatives.csv", encoding='utf-8-sig',index=False)

In [22]:
df_final_owners.to_csv("data/beneficials.csv", encoding='utf-8-sig',index=False)